### ISSUE TITLE
Is it normal to get 0.2 accuracy on CUB using Relation Networks?

### ISSUE THREAD

**Issue by deemoe404** [sicara/easy-few-shot-learning#142]
**Problem**
I start with this Jupyter notebook [Example of classical training](https://github.com/sicara/easy-few-shot-learning/blob/master/notebooks/classical_training.ipynb) . To test out Relation Networks, I change the few-shot classifier to `few_shot_classifier = RelationNetworks(model, feature_dimension=640).to(DEVICE)`, and add `model.use_pooling = False` before validation. But I got `loss=0.253` and `accuracy=0.174` on epoch 200, the best result was `loss=2.94` and `accuracy=0.217` on epoch 39.

By the way, I've change the model state saving code to `best_state = copy.deepcopy(model.state_dict())` (it was `few_shot_classifier.state_dict()`) because I can't get the code running without this change, and it runs fine on Prototypical Networks (70%+ accuracy). But I mention it here in case it's cause by my improper handling of the error.

**Comment by ebennequin** [comment]
`RelationNetworks` use an additional trainable `relation_module` that can only be trained in an episodic fashion. Your process never trains the relation module, thus the random classification.

Note that the weights or specific training procedure for Relation Networks were not made available by the authors. This is why the method wasn't reproduced in our [benchmark](https://github.com/sicara/easy-few-shot-learning/blob/master/easyfsl/modules/resnet.py). In fact, the method is only represented in EasyFSL for completeness. I strongly suggest you use a more proven and reproducible method (see the benchmark).

**Comment by deemoe404** [comment]
Oh, thank you for your response! That explanation resonates with me, and I appreciate the suggestion🥰
